### DISCUSSION TITLE
New Coordinator (new IEEE)

### DISCUSSION THREAD

**Discussion by darkside9009** [Koenkk/zigbee2mqtt#22968] [Devices]
Hello,
I'm fighting since few days to get my new coordinator working with my current setup. I had the Sonof USB (ember), which I want to change to a Zigstr LAN version. When I disconnect the old USB and change the pro to the tcp:, the Z2M won't start because of a panID issue, when I generate a ne panID,  Z2M start all devices are there, but it isn't working. I read in FAQ that I need to change the IEEE (new LAN ZigSTar) to the old Sonof stick. I'm trying to change it, but it seems to be not possible. ZigStar Multi tool don't change the IEEE, TI SmartRF flasher don't want to flash the new hex file, CC2538-bls is flashing but also cannot change the IEEE. My question now is, is it possible to change the coordinator without changing the IEEE? Is there a possibility to change the config files in my Z2M (I have all back ups) I don't want to repairing all my device. There should be a possibility to change the coordinator with a different IEEE or not? What happens when a coordinator gets broken?

**Comment by duncanhancox** [comment]
this post wil guide you through changing coordinator as it has been discussed many times:

https://github.com/Koenkk/zigbee2mqtt/discussions/16478

from your post it looks like you are moving from a silabs chipset (ember) to a TI chipset
zigpy-cli will back up the radio from your old coordinator then restore it to the new TI coordinator, and can also copy the old ieee address to the new coordinator using cc2538-bsl.py but you also have to reflash the firmware at the same time as changing the ieee to make it work:

the command will look like this python3 cc2538-bsl.py -evw --ieee-address xx:xx:xx:xx:xx:xx:xx:xx -p socket://ip:6638 ./CC1352P2_CC2652P_other_coordinator_20230507.hex
replace xx:xx:xx:xx:xx:xx:xx:xx with the your IEEE and replace IP with IP of your SLZB-06

Run ./cc2538-bsl.py -evw --ieee-address 00:12:4b:aa:bb:cc:dd:ee -p /dev/tty.usbserial-10 ./fw.hex, replace:
00:12:4b:aa:bb:cc:dd:ee with your coordinator ieee address (first 0x can be skipped)
/dev/tty.usbserial-10 with the path to your adapter
./fw.hex with the path to your adapters firmware

for a usb stick

you cannot use a new or different ieee without having to repair all devices but you can change the new coordinator ieee to match the old

in the disussion above, there is a solution to the pan id issue towards the bottom

**Reply by darkside9009** [reply]
Hi, thx for your reply. The problem is I have tried with different tools like ZigStarGW-MT, cc2538-bsl and the TI smarter flasher, non of them is able to change my IEEE.
Here is an example of CC2538-bsl:

/Users/st/cc2538-bsl-master/cc2538-bsl.py -evw --ieee-address 90:39:5e:ff:fe:34:fb:b0 -p /dev/tty.usbserial-31230 /Users/st/Downloads/CC1352P2_CC2652P_launchpad_coordinator_20230507.hex

and got this msg in terminal:

Opening port /dev/tty.usbserial-31230, baud 500000 Reading data from /Users/st/Downloads/CC1352P2_CC2652P_launchpad_coordinator_20230507.hex 
Your firmware looks like an Intel Hex file Connecting to target... CC1350 PG2.0 (7x7mm): 704KB Flash, 20KB SRAM, CCFG.BL_CONFIG at 0x000AFFD8 
Primary IEEE Address: 00:12:4B:00:2E:0E:15:15 
Performing mass erase Erasing all main bank flash sectors Erase done 
Writing 360448 bytes starting at address 0x00000000 Write 104 bytes at 0x00057F988 
Write done                                 
Verifying by comparing CRC32 calculations. Verified (match: 0xe83aa727) 
Setting IEEE address to 90:39:5e:ff:fe:34:fb:b0 Writing 8 bytes starting at address 0x000AFFC8 Write 8 bytes at 0x000AFFC8 
Set address done 

it says the the new address is set, but is not set. I have checked the primary address is still old and the secondary always shows FF FF FF FF FF FF FF FF

I'm really fighting since almost one week to change the IEEE, but all options doesn't work. :-(

**Comment by duncanhancox** [comment]
only thing i can think of is using another machine, eg windows, or a fresh setup on a pi - i used a separate pi4 for the flashing tools that doesnt have anything else installed on it

when i used zigpy-cli to backup from a dongle-e (silabs) to a dongle-p (Ti), the zigpy-cli restore also restored the backed up ieee to the dongle-p - worth a try maybe

**Reply by darkside9009** [reply]
I have tried on two different Windows and also Mac.

**Reply by duncanhancox** [reply]
sorry, run out of ideas

**Comment by duncanhancox** [comment]
have you got a spare raspberry pi or similar?

i had trouble changing the ieee on a dongle-e so set up a brand new pi, no zigbee2mqtt or anything, just the various flashing tools and that worked fine just using putty/telnet and ran the tools as root
